### DISCUSSION TITLE
Lesson 9 I keep getting a ```transfer amount exceeds balance``` error

### DISCUSSION THREAD

**Discussion by mrthedude** [Cyfrin/foundry-full-course-cu#683] [General]
I'm currently at [6:39 ](https://youtu.be/sas02qSFZ74?t=23938) into lesson 9 and my ```forge build``` runs fine but when I run ```forge test --fork-url $SEPOLIA_RPC_URL``` I am getting the following error:
![Screenshot 2023-08-21 183952](https://github.com/Cyfrin/foundry-full-course-f23/assets/95384358/067df6e5-dcd1-4e06-8ba7-6cfc197df741)

I tried troubleshooting with ```-vvvvv``` and I think the problem is coming from my ```setup()``` in ```RaffleTest``` causing problems with the other contracts but I'm stuck on how to proceed further to figure it out
![Screenshot 2023-08-21 184502](https://github.com/Cyfrin/foundry-full-course-f23/assets/95384358/438818d0-d7d1-4db0-8e7f-6f8066e24a03)
![Screenshot 2023-08-21 184513](https://github.com/Cyfrin/foundry-full-course-f23/assets/95384358/62bb0375-4de4-404d-b29e-a445d4d85905)

Here's my ```RaffleTest.t.sol``` where I think the problem might be but not sure where...
```
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {DeployRaffle} from "../../script/DeployRaffle.s.sol";
import {Raffle} from "../../src/Raffle.sol";
import {Test, console} from "forge-std/Test.sol";
import {HelperConfig} from "../../script/HelperConfig.s.sol";
import {Vm} from "forge-std/Vm.sol";
import {VRFCoordinatorV2Mock} from "@chainlink/contracts/src/v0.8/mocks/VRFCoordinatorV2Mock.sol";

contract RaffleTest is Test {
    /** Events */
    event EnteredRaffle(address indexed player);

    Raffle public raffle;
    HelperConfig public helperConfig;

    uint256 entranceFee;
    uint256 interval;
    address vrfCoordinator;
    bytes32 gasLane;
    uint64 subscriptionId;
    uint32 callbackGasLimit;
    address link;
    uint256 deployerKey;

    address public PLAYER = makeAddr("player");
    uint256 public constant STARTING_USER_BALANCE = 10 ether;

    function setUp() external {
        DeployRaffle deployer = new DeployRaffle();
        (raffle, helperConfig) = deployer.run();
        vm.deal(PLAYER, STARTING_USER_BALANCE);
        (
            entranceFee,
            interval,
            vrfCoordinator,
            gasLane,
            subscriptionId,
            callbackGasLimit,
            link,

        ) = helperConfig.activeNetworkConfig();
    }
```
I left out the rest of the test functions but please let me know if you need to see them 🙏 🙏 

Appreciate any help!
thankyou!

**Comment by theirrationalone** [comment]
The traces show that you have some problem with `LinkToken` in your helperConfig, Please Verify your implementation again...!

**Reply by mrthedude** [reply]
weird, just tried copying and pasting the repo contract code for the ```LinkToken.sol``` and ```HelperConfig.s.sol``` because I still can't find the issue and it's still showing the same error... i'm lost on this one lol

**Reply by mrthedude** [reply]
solved it, i think it was from not having a chainlink vrf subscription

**Reply by theirrationalone** [reply]
Woah! Great work @mrthedude !

Now you can check all events out by going to etherscan -> transactions, when you spit out transactions to sepolia Or goerli... Go let's have a look. If subscription is working well.
